### ISSUE TITLE
Autotools script does not pass `-lltld` to the linker.

### ISSUE THREAD

**Issue by ghost** [opencog/guile-dbi#7]
See https://gitlab.com/Lockywolf/lwfslackbuilds/-/blob/master/61_artanis_deps/guile-dbi/guile-dbi.SlackBuild

I have to pass `-lltdl` to CFLAGS manually. This should not be the case. In fact, most packagers for linux distros do it like this, however, `configure.ac` has some code to deal with it, but commented-out.

**Comment by linas** [comment]
> commented-out.

Where?  Perhaps it should be uncommented?

**Comment by ghost** [comment]
I have not idea. Line 60 and nearby of `configure.ac`. No idea, I never worked with autotools.

**Comment by linas** [comment]
Ah!  Line 62 is interesting, too, it might solve your other problem. Edit `configure.ac`, uncomment lines 60,61,62 and run `./autogen.conf --no-configure` to regenerate the `configure` script. Then run `configure` in the way you desire. I suspect this will resolve both of your problems.

**Comment by linas** [comment]
h, and let me know if it works. I will uncomment if it works for you.

**Comment by linas** [comment]
I just tried w/ your config and its not working. Debugging now

**Comment by linas** [comment]
I just pushed commit 56e12dcab139c373dafccdd72fa2a140d82f3910 which, from what I can tell, fixes both #6 and #7. Pull, rebuild, be sure to run `autogen.sh` first.

**Comment by linas** [comment]
I just merged pull req #8 which partly undoes the work of https://github.com/opencog/guile-dbi/commit/56e12dcab139c373dafccdd72fa2a140d82f3910 which was claimed to fix this issue. In addition, I am now seeing this message:
```
checking for guile-3.0... yes
../configure: line 20324: GUILE_PKG: command not found
../configure: line 20325: GUILE_FLAGS: command not found
../configure: line 20326: GUILE_SITE_DIR: command not found
```
which says that the other half of that patch does nothing at all. which implies that all of the reasoning that went in to fix this bug is flawed or somehow unreproducible, or, most likely, distro-dependent. I'm confused.

**Comment by linas** [comment]
An hour later, turns out I needed to say
```
ACLOCAL_PATH=/usr/local/share/aclocal/ ./autogen.sh --no-configure
```

Maybe that's what you needed as well. Closing this issue as 'stale' If there are still problems, open a new issue.
